validate.go

```diff
@@ -18,34 +37,83 @@ func validate(payload []byte) ([]byte, error) {
 
 	logger.Info("validating request")
 
+	if kind := gjson.GetBytes(payload, "request.object.kind"); kind.String() != "Pod" {
+		return kubewarden.RejectRequest(
+			kubewarden.Message("object is not of kind Pod: rejecting request"),
+			kubewarden.Code(421))
+	}
+
 	data := gjson.GetBytes(
 		payload,
-		"request.object.metadata.name")
+		"request.object.spec.securityContext.sysctls")
 
 	if !data.Exists() {
-		logger.Warn("cannot read object name from metadata: accepting request")
+		logger.Warn("pod doesn't specify sysctls: accepting request")
 		return kubewarden.AcceptRequest()
 	}
-	name := data.String()
 
-	logger.DebugWithFields("validating ingress object", func(e onelog.Entry) {
-		namespace := gjson.GetBytes(payload, "request.object.metadata.namespace").String()
+	logger.DebugWithFields("validating pod object", func(e onelog.Entry) {
+		name := gjson.GetBytes(payload, "request.object.metadata.name").String()
+		namespace := gjson.GetBytes(payload,
+			"request.object.metadata.namespace").String()
 		e.String("name", name)
 		e.String("namespace", namespace)
 	})
 
-	if settings.DeniedNames.Contains(name) {
-		logger.InfoWithFields("rejecting ingress object", func(e onelog.Entry) {
+	// build set of safe sysctls:
+	knownSafeSysctlsArray := getSafeSysctls()
+	knownSafeSysctls := mapset.NewThreadUnsafeSet()
+	for i := 0; i < len(knownSafeSysctlsArray); i++ {
+		knownSafeSysctls.Add(knownSafeSysctlsArray[i])
+	}
+
+	// build set of prefixes from patterns of forbidden sysctls:
+	globForbiddenSysctls := mapset.NewThreadUnsafeSet()
+	for _, elem := range settings.ForbiddenSysctls.ToSlice() {
+		if strings.HasSuffix(elem.(string), "*") {
+			globForbiddenSysctls.Add(strings.TrimSuffix(elem.(string), "*"))
+		}
+	}
```

Comment:
I would drop/move this code somewhere else. 

I would change the internal implementation of Settings to hold:
  * `ForbiddenSysctls`
  * `ForbiddenGlobSysctls`
  * `AllowedUnsafeSysctls`

I think we could also add the `getSafeSysctls` sysctls into `AllowedUnsafeSysctls`

Reply:
A priori sounds like a good idea at least for clarity. But then the Settings struct will not match the json (nice if we go with encoding/json). We could built an internal struct with the `forbidden`, `forbiddenGlob`, `safe` sets, instead of creating them at the beginning of evaluation. But I prefer to leave the current `knownSafeSysctls` because then We can provide a better message when rejecting.

Simplified `knownSafeSysctls` creation by using a set instead of array in https://github.com/kubewarden/psp-sysctl-policy/pull/1/commits/8b03dfdf2592e8e5c2bc6b6c32ba929bde7bdb61 (I was using an array to signify that they are "constant", but it's not really needed).

I don't think it leads to a faster implementation though; right now we are iterating through the pod sysctls and if we find one sysctl that is not good, we stop and reject. With the sets, we would first perform set operations (which, given that the are implemented with maps is going to need the same iterations), and then iterate nevertheless.